rudder-core/src/main/scala/com/normation/rudder/migration/XmlEntityMigration.scala

```diff
@@ -127,135 +163,139 @@ trait ControlXmlFileFormatMigration extends XmlFileFormatMigration {
-          migrator.migrate()
-        }
-        this.migrate()
-
-
-      //other case: does nothing
-      case Some(x) =>
-        logger.debug(s"Migration of EventLog from format '${fromVersion}' to '${toVersion}': nothing to do")
-        Full(MigrationVersionNotHandledHere)
-    }
+          }
+
+
+           boxSequence(migrationResults) match {
+            case Full(seq) =>
+              val numberMigrated = seq.collect { case MigrationSuccess(i) => i }.sum
+              migrationEventLogRepository.setMigrationFileFormat(id, toVersion, DateTime.now)
+              logger.info(s"Completed migration to file format '${toVersion}', ${numberMigrated} records migrated")
+              Full(MigrationSuccess(numberMigrated))
+            case eb:EmptyBox => eb
+          }
+
+        //a past migration was done, but the final format is not the one we want
+        case \/-(Some(x@DB.MigrationEventLog(
+            _
+          , _
+          , _
+          , _
+          , Some(endTime)
+          , Some(migrationFileFormat)
+          , _
+        ))) if(migrationFileFormat == fromVersion) =>
+          //create a new status line with detected format = migrationFileFormat,
+          //and a description to say why we recurse
+          migrationEventLogRepository.createNewStatusLine(migrationFileFormat, Some(s"Found a post-migration fileFormat='${migrationFileFormat}': update"))
+          this.migrate()
+
+            // lower file format found, send to parent)
+        case \/-(Some(status@DB.MigrationEventLog(
+            _
+          , _
+          , detectedFileFormat
+          , migrationStartTime
+          , migrationEndTime : None.type
+          , _
+          , _
+        ))) if(detectedFileFormat < fromVersion) =>
+
+
+          logger.info("Found and older migration to do")
+          previousMigrationController match {
+            case None =>
+              logger.info(s"The detected format ${detectedFileFormat} is no more supported, you will have to " +
+                  "use an installation of Rudder that understand it to do the migration. For information, " +
+                  "Rudder 2.6.x is the last major version which is able to import file format 1.0")
+              Full(MigrationVersionNotSupported)
+
+            case Some(migrator) => migrator.migrate() match{
+              case Full(MigrationSuccess(i)) =>
+                  logger.info("Older migration completed, relaunch migration")
+                  this.migrate()
+              case eb:EmptyBox =>
+                  val e = (eb ?~! s"Older migration failed, Could not correctly finish the migration from EventLog fileFormat from '${fromVersion}' to '${toVersion}'. Check logs for errors. The process can be trigger by restarting the application")
+                  logger.error(e)
+                  e
+              case _ =>
+                  logger.info("Older migration completed, relaunch migration")
```

Comment:
We might want to clarify what is the 'Older migration', it may confuse the user ?